textdb/textdb-dataflow/src/test/java/edu/uci/ics/textdb/dataflow/keywordmatch/KeywordMatcherTest.java

```diff
@@ -0,0 +1,402 @@
+        dataWriter.clearData();
+    }
+
+    /**
+     * Tokenizes the query string using the given analyser
+     * @param analyzer
+     * @param query
+     * @return ArrayList<String> list of results
+     */
+    public ArrayList<String> tokenizeQuery(Analyzer analyzer,  String query) {
+        HashSet<String> resultSet = new HashSet<>();
+        ArrayList<String> result = new ArrayList<String>();
+        TokenStream tokenStream  = analyzer.tokenStream(null, new StringReader(query));
+        CharTermAttribute charTermAttribute = tokenStream.addAttribute(CharTermAttribute.class);
+
+        try{
+            tokenStream.reset();
+            while (tokenStream.incrementToken()) {
+                String term = charTermAttribute.toString();
+                resultSet.add(term);
+            }
+            tokenStream.close();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        result.addAll(resultSet);
+
+        return result;
+    }
+
+    /**
+     * Creates a Query object as a boolean Query on all attributes
+     * Example: For creating a query like
+     * (TestConstants.DESCRIPTION + ":lin" + " AND " + TestConstants.LAST_NAME + ":lin")
+     * we provide a list of AttributeFields (Description, Last_name) to search on and a query string (lin)
+     *
+     * BooleanQuery() is deprecated. In future a better solution could be worked out in Query builder layer
+     *
+     * @param query
+     * @param attributeList
+     * @return Query
+     * @throws ParseException
+     */
+    private Query createQueryObject(String query, ArrayList<Attribute> attributeList) throws ParseException {
+        Analyzer analyzer = new StandardAnalyzer();
+        ArrayList<String> tokens;
+        String[] fields = new String[attributeList.size()];
+        for (int i=0;i< attributeList.size();i++){
+            fields[i] = attributeList.get(i).getFieldName();
+        }
+
+        tokens = tokenizeQuery(analyzer, query);
+        BooleanQuery booleanQuery = new BooleanQuery();
+        MultiFieldQueryParser parser = new MultiFieldQueryParser(fields, analyzer);
+        for(String searchToken: tokens){
+            Query termQuery = parser.parse(searchToken);
+            booleanQuery.add(termQuery, BooleanClause.Occur.MUST);
+        }
+        return booleanQuery;
+    }
+
+
+    /**
+     * For a given string query & list of attributes it gets a list of results
+     * buildMultiQueryOnAttributeList flag decides if the query is formed as a boolean Query on all attribute
+     * or all records are scanned
+     * @param query
+     * @param attributeList
+     * @param buildMultiQueryOnAttributeList
+     * @return List<ITuple>
+     * @throws DataFlowException
+     * @throws ParseException
+     */
+
+    public List<ITuple> getQueryResults(String query, ArrayList<Attribute> attributeList, boolean buildMultiQueryOnAttributeList) throws DataFlowException, ParseException {
```

Comment:
Since this test case uses "ATTRIBUTES_PEOPLE", it may be better to include it in the function name as "getPeopleQueryResults"?